rest/src/main/kotlin/request/StackTraceRecoveringKtorRequestHandler.kt

```diff
@@ -1,48 +1,48 @@
 package dev.kord.rest.request
 
+import kotlin.DeprecationLevel.ERROR
 
 /**
  * Extension of [KtorRequestHandler] which tries to recover stack trace information lost through Ktor's
  * [io.ktor.util.pipeline.SuspendFunctionGun].
  *
- * This is done by creating a [ContextException] to capture the stack trace up until the point just before
- * [KtorRequestHandler.handle] gets called, then if that call throws any type of [Throwable] the [ContextException] gets
- * thrown instead (See [ContextException.cause])
+ * This is done by creating a [RecoveredStackTrace] to capture the stack trace up until the point just before
+ * [KtorRequestHandler.handle] gets called, then if that call throws any type of [Throwable] the [RecoveredStackTrace]
+ * gets added to the original [Throwable] as a [suppressed exception][addSuppressed] and the original [Throwable] is
+ * rethrown.
  *
- * @see ContextException
  * @see withStackTraceRecovery
  */
 public class StackTraceRecoveringKtorRequestHandler(private val delegate: KtorRequestHandler) :
     RequestHandler by delegate {
 
-    /**
-     * @throws ContextException if any exception occurs (this is also the only exception which can be thrown)
-     * @see KtorRequestHandler.handle
-     */
+    /** @see KtorRequestHandler.handle */
     override suspend fun <B : Any, R> handle(request: Request<B, R>): R {
-        val stacktrace = ContextException()
+        val recoveredStackTrace = RecoveredStackTrace()
 
         return try {
             delegate.handle(request)
-        } catch (e: Exception) {
-            stacktrace.sanitizeStackTrace()
-            e.initCause(stacktrace)
+        } catch (e: Throwable) {
```

Comment:
Should we really add the recovered stack trace to all `Throwable`s? Or should it just be `Exception`s?

Reply:
At the end of the day it doesn't really matter.

I think that catching all `Throwable`s at least has the advantage that if a weird exception happens (example: `LinkageError`) at least we know where it is being thrown if stacktrace recovery is enabled.

Reply:
> I think that catching all `Throwable`s at least has the advantage that if a weird exception happens (example: `LinkageError`) at least we know where it is being thrown if stacktrace recovery is enabled.

Was thinking something similar 👍